### ISSUE TITLE
Suggestion : Make frame data fade in and fade out

### ISSUE THREAD

**Issue by manishguptaji** [WAZAAAAA0/TekkenBot#36]
Make frame data fade in and fade out as done by [https://github.com/TekkenOverlay/TekkenOverlay#tekken-overlay](url)
Instead of showing fixed frame data at one place, make it float with the player which makes it very very easy to see and react too. 

Since Tekken Overlay introduced the new [overlay] prefix with user name most of the people wont use it now and will use your overlay. So it will help them a lot.

**Comment by WAZAAAAA0** [comment]
that's pretty much out of scope for this project, it would require a whole rewrite of the thing
TBP doesn't handle DirectX injections at all, which is why it doesn't work on exclusive full screen mode... while for TO it's the main focus in order to draw hitboxes

you'll have more luck reversing the name-changer feature somehow lol

**Comment by manishguptaji** [comment]
"you'll have more luck reversing the name-changer feature somehow lol"
only if it was open source

**Comment by WAZAAAAA0** [comment]
I hear ya broski.
it's not like you can't use it, it just changes your name when you do
Anyway I tried reversing it for fun, it should bypass the whole [overlay] name changer thing and behave as usual, see how that works for you @manishguptaji 
[tekken-overlayNAMELESS.zip](https://github.com/WAZAAAAA0/TekkenBot/files/6236832/tekken-overlayNAMELESS.zip)

**Comment by manishguptaji** [comment]
> > only if it was open source
> 
> I hear ya broski.
> it's not like you can't use it, it just changes your name when you do
> Anyway I tried reversing it for fun, it should bypass the whole [overlay] name changer thing and behave as usual, see how that works for you @manishguptaji
> [tekken-overlayNAMELESS.zip](https://github.com/WAZAAAAA0/TekkenBot/files/6236832/tekken-overlayNAMELESS.zip)

Thanks man it worked :D

**Comment by schinsue** [comment]
@WAZAAAAA0 Out of interest, what program did you use to reverse engineer the dll? Also want to do it for fun myself.

**Comment by muzzone** [comment]
> > only if it was open source
> 
> I hear ya broski.
> it's not like you can't use it, it just changes your name when you do
> Anyway I tried reversing it for fun, it should bypass the whole [overlay] name changer thing and behave as usual, see how that works for you @manishguptaji
> [tekken-overlayNAMELESS.zip](https://github.com/WAZAAAAA0/TekkenBot/files/6236832/tekken-overlayNAMELESS.zip)

Еhank you very much :)

**Comment by WAZAAAAA0** [comment]
THE UNBLOCKABLE https://github.com/TekkenOverlay/TekkenOverlay/issues/44

**Comment by WAZAAAAA0** [comment]
> @WAZAAAAA0 Out of interest, what program did you use to reverse engineer the dll? Also want to do it for fun myself.

@schinsue Cheat Engine. This is how I did it, it's pretty ghetto but Hey, as long as it works. TL;DR version at the bottom

- with the DLL overlay attached to a match, scan the string [overlay]
- find the one nickname address actually monitored by the overlay by changing one by one until the overlay stops working ("Overlay disabled until name change syncs up bla bla")
- now we have an address that we can use to play around and quickly find the more important "IsOverlayEnabled" address (not a real name I just made it up). I did it with these scan settings: "Unknown initial value - Byte - tekken-overlay.dll" but spoiler alert: 0=OFF 1=ON ("tekken-overlay.dll"+F7AE0)
- right click the newly found IsOverlayEnabled -> Find out what writes to this address -> Add to the code list -> Open the disassembler at this location -> replace the CALL that's right above it with NOP's, now the overlay should stay permanently ON (you just removed the logic that makes it ever go OFF)
- take note of the hex bytes you just changed, so you can apply them permanently with HxD Hex Editor and save the final modified DLL

TL;DR with CE find the address that decides if the overlay is ON/OFF in real-time, find the opcode that writes to it, jump to that memory area, fix the assembly code to kill the name changer function, save the changes to the actual DLL file with any hex editor

**Comment by schinsue** [comment]
@WAZAAAAA0 Thanks for your instructions, meanwhile I managed to reverse engineer it as well btw. How can you open the disassembler at a specific memory location?

I used x64dbg and Cheat Engine. I was going to make a tool to change the name back. I found the address that was responsible for changing names and then found out about the message popping up. Used x64dbg to find the string and read/changed the assembly code to jump the function. Could create a patched DLL straight from x64dbg after that.

As this is my first time doing anything reverse engineering, I even tried to get rid of the pesky message about the overlay disabling and was successful. Will try to make a tool in Rust to change the name just to practice a bit. 

I can post my DLL, but I think the creator doesn't like us editing the functionality. So I won't for now as it was just for fun for me.

Thanks again btw, this was a lot of fun.

**Comment by WAZAAAAA0** [comment]
> How can you open the disassembler at a specific memory location?

Advanced Options > double click the lonely address in the list to bring you there
or, if the list is empty and you just want to jump to an arbitrary location: Memory View -> Ctrl+G (Goto Address)

**Comment by schinsue** [comment]
Thanks, I looked at it. Pretty useful. Now trying to make a tool to change the name by changing the memory address directly in Tekken. Trying to use any excuse I can find to try and use Rust somewhere 😄

**Comment by MeguminYun** [comment]
@WAZAAAAA0 I have followed the steps and am able to get the same results in cheat engine.
Follow their most recent update, i found that their new address responsible for the "IsOverlayEnabled" is "tekken-overlay.dll"+F9B80. This lead me to the Call function right above to change bytes E8 DD4E0100 call tekken-overlay.dll+30800 to NOP.
The trouble im having is that im unable to goto this address or find this set of bytes in 64xdbg.

How do i apply changes i found in CE to the hex editor ?
![Untitled](https://user-images.githubusercontent.com/81469496/120019367-f2caed00-c01a-11eb-8313-cf2e41abd0e0.png)

**Comment by muzzone** [comment]
@MeguminYun please let me know if you manage to crack it

**Comment by WAZAAAAA0** [comment]
@MeguminYun @muzzone you don't really need to bother reversing it again really, just use the previously bypassed 4.10 version as a base, and copy the newer 4.20 memory addresses into it lolz (`memory-addresses.yaml` + `memory-addresses-cache.yaml`)

**Comment by MeguminYun** [comment]
@WAZAAAAA0 hmm there must be something different then, because it didn't work despite being the first thing I tried. Thus deciding to try reversing it.
I did it because I assumed it works similarly to TBP by reading the memory pointers.

**Comment by WAZAAAAA0** [comment]
@MeguminYun works for me ¯\\\_(ツ)_/¯ 
![4.20](https://i.imgur.com/FhMdLuV.png)
this is what I used, try this maybe: [TekkenOverlayNAMELESSmix.zip](https://github.com/WAZAAAAA0/TekkenBot/files/6565798/TekkenOverlayNAMELESSmix.zip)

**Comment by MeguminYun** [comment]
@WAZAAAAA0 ah this works perfectly for me. Was the only thing that you changed the memory-addresses.yaml and memory-addresses-cache.yaml from the 4.2 release ?

**Comment by WAZAAAAA0** [comment]
> Was the only thing that you changed

the second difference was the old "nameless bypass" from `tekken-overlayNAMELESS.zip` (`E8 6D 42 01 00`->`90 90 90 90 90`)

**Comment by muzzone** [comment]
@WAZAAAAA0 it works! thanks :)

**Comment by MeguminYun** [comment]
@WAZAAAAA0 old nameless bypass ? Is it the original bypass method found through cheat engine ?

Also, I do appreciate you taking the time to help with bypassing another similar project. Wanted to learn more about reverse engineering. Thank you !

**Comment by WAZAAAAA0** [comment]
> old nameless bypass ? Is it the original bypass method found through cheat engine ?

I mean the zip I uploaded here `tekken-overlayNAMELESS.zip` https://github.com/WAZAAAAA0/TekkenBot/issues/36#issuecomment-811060700

**Comment by hhm-h** [comment]
> > @WAZAAAAA0 Out of interest, what program did you use to reverse engineer the dll? Also want to do it for fun myself.
> 
> @schinsue Cheat Engine. This is how I did it, it's pretty ghetto but Hey, as long as it works. TL;DR version at the bottom
> 
> * with the DLL overlay attached to a match, scan the string [overlay]
> * find the one nickname address actually monitored by the overlay by changing one by one until the overlay stops working ("Overlay disabled until name change syncs up bla bla")
> * now we have an address that we can use to play around and quickly find the more important "IsOverlayEnabled" address (not a real name I just made it up). I did it with these scan settings: "Unknown initial value - Byte - tekken-overlay.dll" but spoiler alert: 0=OFF 1=ON ("tekken-overlay.dll"+F7AE0)
> * right click the newly found IsOverlayEnabled -> Find out what writes to this address -> Add to the code list -> Open the disassembler at this location -> replace the CALL that's right above it with NOP's, now the overlay should stay permanently ON (you just removed the logic that makes it ever go OFF)
> * take note of the hex bytes you just changed, so you can apply them permanently with HxD Hex Editor and save the final modified DLL
> 
> TL;DR with CE find the address that decides if the overlay is ON/OFF in real-time, find the opcode that writes to it, jump to that memory area, fix the assembly code to kill the name changer function, save the changes to the actual DLL file with any hex editor

Hello, brother.

After reading this article above, I thought I wanted to try it.

So I studied a lot and recognized it, but I didn't succeed. That's why I want to ask my brother.

Could you tell me the explanation above in detail?

My brother may be busy, but I ask you.

**Comment by WAZAAAAA0** [comment]
@hhm-h that's as many details as I could give ¯\\\_(ツ)_/¯ if the explanation is too hard to follow, learning the basics of CE might help to understand them better:
> download CE from cheatengine.org and do the tutorial bundled within the installation (`Tutorial-i386.exe`), the first 3 steps should teach you enough and shouldn’t take longer than 10 minutes to complete.

if you're telling me you correctly followed that explanation and it still didn't work, it's probably **outdated** and stopped working like @MeguminYun said.

...just ignore everything and use this, it works for now https://github.com/WAZAAAAA0/TekkenBot/files/6565798/TekkenOverlayNAMELESSmix.zip

**Comment by hhm-h** [comment]
> @ hhm-h 설명을 따르기가 너무 어려우면 ¯ \ _ (ツ) _ / ¯ 제공 할 수있는 자세한 내용입니다. CE의 기본 사항을 배우는 것이 더 잘 이해하는 데 도움이 될 수 있습니다.
> 
> > cheatengine.org에서 CE를 다운로드하고 설치 ( `Tutorial-i386.exe`)에 번들로 제공되는 자습서를 수행하십시오. 처음 3 단계는 충분히 가르쳐야하며 완료하는 데 10 분 이상 걸리지 않아야합니다.
> 
> 당신이 나를 이야기하는 경우가 제대로 그 설명을 따라 여전히 일을하지 않았다, 그것은 아마 **구식** 과 같은 작동이 중지 @MeguminYun는 여기 말했다 [# 36 (코멘트)](https://github.com/WAZAAAAA0/TekkenBot/issues/36#issuecomment-850556675) .
> 
> ... 모든 것을 무시하고 이것을 사용하십시오. 지금은 작동합니다 https://github.com/WAZAAAAA0/TekkenBot/files/6565798/TekkenOverlayNAMELESSmix.zip

Thank you for replying.
I wanted to study this for fun, just like the other brothers up there. I'll study more and try it. Thank you so much for replying to me, brother!
I always respect and appreciate my brothers. I hope you have a good day today!

**Comment by WAZAAAAA0** [comment]
been a while, putting out an updated version of this due to request [TekkenOverlay-4.27.2-.2022-02-19NAMELESS.zip](https://github.com/WAZAAAAA0/TekkenBot/files/8450614/TekkenOverlay-4.27.2-.2022-02-19NAMELESS.zip)

**Comment by rajnmaker** [comment]
> been a while, putting out an updated version of this due to request [TekkenOverlay-4.27.2-.2022-02-19NAMELESS.zip](https://github.com/WAZAAAAA0/TekkenBot/files/8450614/TekkenOverlay-4.27.2-.2022-02-19NAMELESS.zip) @rajnmaker #44 gotchu covered, see if that works [TekkenOverlay-5.00.0-.2022-08-18NAMELESS.zip](https://github.com/WAZAAAAA0/TekkenBot/files/9378933/TekkenOverlay-5.00.0-.2022-08-18NAMELESS.zip)

Thank you so much!

**Comment by rajnmaker** [comment]
it works!

**Comment by NEURTX** [comment]
> некоторое время выпускал обновленную версию по запросу [TekkenOverlay-4.27.2-.2022-02-19NAMELESS.zip](https://github.com/WAZAAAAA0/TekkenBot/files/8450614/TekkenOverlay-4.27.2-.2022-02-19NAMELESS.zip)
> 
> @rajnmaker [# 44](https://github.com/WAZAAAAA0/TekkenBot/issues/44) попался, посмотрите, работает ли это [TekkenOverlay-5.00.0-.2022-08-18NAMELESS.zip](https://github.com/WAZAAAAA0/TekkenBot/files/9378933/TekkenOverlay-5.00.0-.2022-08-18NAMELESS.zip)

there is the same overlay, only version 5.10? you can send here.

**Comment by WAZAAAAA0** [comment]
@NEURTX ok [TekkenOverlay-5.10.1-.2022-12-16-NoNametagNoWatermark.zip](https://github.com/WAZAAAAA0/TekkenBot/files/10250485/TekkenOverlay-5.10.1-.2022-12-16-NoNametagNoWatermark.zip)

**Comment by NEURTX** [comment]
> @NEURTXхорошо [TekkenOverlay-5.10.1-.2022-12-16-NoNametagNoWatermark.zip](https://github.com/WAZAAAAA0/TekkenBot/files/10250485/TekkenOverlay-5.10.1-.2022-12-16-NoNametagNoWatermark.zip)

hi, do you have TekkenOverlay-5.11.5-[2023-05-01]? you can send. THX

**Comment by WAZAAAAA0** [comment]
@NEURTX try [TekkenOverlay-5.11.5-.2023-05-01-NoNametagNoWatermark.zip](https://github.com/WAZAAAAA0/TekkenBot/files/11411743/TekkenOverlay-5.11.5-.2023-05-01-NoNametagNoWatermark.zip)

@Golundo RE: https://github.com/TekkenOverlay/TekkenOverlay/issues/118
this was the modification:
> Just replace the relevant text with spaces using any hex editor yourself.
in `tekken-overlay.dll` search for:
`25 73 20 76 25 73 20 5B 25 73 5D`
replace with:
`20 20 20 20 20 20 20 20 20 20 20`

or you can use the pre-modified file I just linked

**Comment by Golundo** [comment]
Legend! was looking for this. Much appreciated!

**Comment by Hasekoni** [comment]
Hi do you have the last build with no nametag?

**Comment by WAZAAAAA0** [comment]
@Hasekoni ait [TekkenOverlay-5.11.8-.2023-06-16-NoNametagNoWatermark.zip](https://github.com/WAZAAAAA0/TekkenBot/files/12643944/TekkenOverlay-5.11.8-.2023-06-16-NoNametagNoWatermark.zip)

**Comment by Hasekoni** [comment]
It is possible to do it for Tekken 8 release?

**Comment by WAZAAAAA0** [comment]
...is there even anything to do with it? does that even have nametag features yet?

**Comment by Yozow** [comment]
It has the same distracting watermark line at the bottom while playing that you removed in Tekken 7

**Comment by J-Pedro-Nogueira** [comment]
- Open Tekken8Overlay.dll with any hex editor
- Search for 43 80 3C 02 00 (it's at line 5F40)
- Replace those values with a NOP (90 90 90 90 90)
- That's it, I'm basically just parroting @WAZAAAAA0's instructions

It seems to work fine but I had never used CE to find which instructions are being called in real time. Either wait for someone with more experience or use at your own risk.

**Comment by WAZAAAAA0** [comment]
> * Open Tekken8Overlay.dll with any hex editor
> 
>     * Search for 43 80 3C 02 00 (it's at line 5F40)
> 
>     * Replace those values with a NOP (90 90 90 90 90)
> 
>     * That's it, I'm basically just parroting @WAZAAAAA0's instructions
> 
> 
> It seems to work fine but I had never used CE to find which instructions are being called in real time. Either wait for someone with more experience or use at your own risk.

@thecoolestlol @CAPTAlNHOWDY *"Both of the threads about this have no real answer"* because the answer got deleted by some repository owner. But if you follow ^ that mini guide you can hide the watermark 👌

**Comment by WAZAAAAA0** [comment]
> * Open Tekken8Overlay.dll with any hex editor
> 
>     * Search for 43 80 3C 02 00 (it's at line 5F40)
> 
>     * Replace those values with a NOP (90 90 90 90 90)
> 
>     * That's it, I'm basically just parroting @WAZAAAAA0's instructions
> 
> 
> It seems to work fine but I had never used CE to find which instructions are being called in real time. Either wait for someone with more experience or use at your own risk.

@thecoolestlol @CAPTAlNHOWDY https://github.com/TekkenOverlay/TekkenOverlay/issues/146 *"Both of the threads about this have no real answer"* because the answer got deleted by some repository owner. But if you follow ^ that mini guide you can hide the watermark 👌
also you can add `NotificationWasAlreadyShown: true` at the bottom of `Tekken8OverlaySettings.yaml` to prevent the fullscreen donation nag from appearing

**Comment by thecoolestlol** [comment]
> > * Open Tekken8Overlay.dll with any hex editor
> >   
> >   * Search for 43 80 3C 02 00 (it's at line 5F40)
> >   * Replace those values with a NOP (90 90 90 90 90)
> >   * That's it, I'm basically just parroting @WAZAAAAA0's instructions
> > 
> > It seems to work fine but I had never used CE to find which instructions are being called in real time. Either wait for someone with more experience or use at your own risk.
> 
> @thecoolestlol @CAPTAlNHOWDY [TekkenOverlay/TekkenOverlay#146](https://github.com/TekkenOverlay/TekkenOverlay/issues/146) _"Both of the threads about this have no real answer"_ because the answer got deleted by some repository owner. But if you follow ^ that mini guide you can hide the watermark 👌 also you can add `NotificationWasAlreadyShown: true` at the bottom of `Tekken8OverlaySettings.yaml` to prevent the fullscreen donation nag from appearing

Thank you <3

**Comment by J-Pedro-Nogueira** [comment]
Someone messaged me saying this doesn't work anymore — it works as long as you edit the new DLL file. The exact fix may change depending on the build, so the best thing to do is to use CE and find the correct instructions every time TekkenOverlay gets updated. Here's the current fix (same change, just on a different address):

https://github.com/J-Pedro-Nogueira/public-repo/releases/tag/v1.0.0

**Comment by raebordz** [comment]
> Someone messaged me saying this doesn't work anymore — it works as long as you edit the new DLL file. The exact fix may change depending on the build, so the best thing to do is to use CE and find the correct instructions every time TekkenOverlay gets updated. Here's the current fix (same change, just on a different address):
> 
> https://github.com/J-Pedro-Nogueira/public-repo/releases/tag/v1.0.0

There's a new update and I cant no longer found the 43 80 3C 02 00 address. Mind helping out?

**Comment by J-Pedro-Nogueira** [comment]
I hadn't used CE in 2 years so I'm definetly out of practice now.... I threw some NOPs around and the most I could manage was corrupting the text being shown. The easiest way now could be replacing the text itself with white space. Everything is easy to find except for the commas and brackets. If you find the correct edit please share it with us

**Comment by J-Pedro-Nogueira** [comment]
> > Someone messaged me saying this doesn't work anymore — it works as long as you edit the new DLL file. The exact fix may change depending on the build, so the best thing to do is to use CE and find the correct instructions every time TekkenOverlay gets updated. Here's the current fix (same change, just on a different address):
> > https://github.com/J-Pedro-Nogueira/public-repo/releases/tag/v1.0.0
> 
> There's a new update and I cant no longer found the 43 80 3C 02 00 address. Mind helping out?

Not gonna bother fiddling with it too much since I dropped the game, but technically ```E9 FC 00 00 00``` @0008A970 works, It stops *all* text from showing so it's not a clean fix, but if you only want the FPS cap for example it should be fine.

**Comment by raebordz** [comment]
> > > Someone messaged me saying this doesn't work anymore — it works as long as you edit the new DLL file. The exact fix may change depending on the build, so the best thing to do is to use CE and find the correct instructions every time TekkenOverlay gets updated. Here's the current fix (same change, just on a different address):
> > > https://github.com/J-Pedro-Nogueira/public-repo/releases/tag/v1.0.0
> > 
> > 
> > There's a new update and I cant no longer found the 43 80 3C 02 00 address. Mind helping out?
> 
> Not gonna bother fiddling with it too much since I dropped the game, but technically `E9 FC 00 00 00` @0008A970 works, It stops _all_ text from showing so it's not a clean fix, but if you only want the FPS cap for example it should be fine.

Yes this works, thanks. Make sure you edit the settings before doing this because it completely removes the text in the UI. Im not good at this things lol

**Comment by J-Pedro-Nogueira** [comment]
Funily enough you can actually edit the settings like normal it's just completely invisible lol. Good enough for typing in the new FPS cap and clicking save I guess

**Comment by maylique** [comment]
> Funily enough you can actually edit the settings like normal it's just completely invisible lol. Good enough for typing in the new FPS cap and clicking save I guess

if i replace the start with E9 FC 00 00 00 @0008A970, the game ist just crashing. am i just stupid? first time trying some hex edit, i managed to remove all except for the [ which i can not remove except in CE.

**Comment by maylique** [comment]
> > Funily enough you can actually edit the settings like normal it's just completely invisible lol. Good enough for typing in the new FPS cap and clicking save I guess
> 
> if i replace the start with E9 FC 00 00 00 @0008A970, the game ist just crashing. am i just stupid? first time trying some hex edit, i managed to remove all except for the [ which i can not remove except in CE.

nvm just add it @0008A971
